cluster_test.go

```diff
@@ -46,6 +46,98 @@ func TestClusterClient(t *T) {
 	assert.Implements(t, new(Client), c)
 }
 
+func makeFailedFlagMap(addrs []string) map[string]bool {
+	failedAddrsFlag := make(map[string]bool)
+	for _, addr := range addrs {
+		failedAddrsFlag[addr] = true
+	}
+	return failedAddrsFlag
+}
+
+func TestClusterInitSync(t *T) {
+	scl := newStubCluster(testTopo)
+	serverAddrs := scl.addrs()
+	{
+		c := scl.newCluster()
+		err := c.Sync(true, false, true)
+		assert.NotNil(t, err)
+	}
+	//part of the addresses are unavailable during the initialization
+	//and recover after that, call Sync to test whether it can work
+	{
+		c, err := scl.newInitSyncErrorCluster(serverAddrs,
+			makeFailedFlagMap(serverAddrs[0:len(serverAddrs)/2]),
+			ClusterWithInitSyncSilent(true))
+		require.Nil(t, err)
+		defer c.Close()
+		//TestClusterSync
+		assertClusterState := func() {
+			require.Nil(t, c.Sync())
+			c.l.RLock()
+			defer c.l.RUnlock()
+			assert.Equal(t, c.topo, scl.topo())
+			assert.Len(t, c.pools, len(c.topo))
+			for _, node := range c.topo {
+				assert.Contains(t, c.pools, node.Addr)
+			}
+		}
+		assertClusterState()
+
+		// cluster is unstable af
```

Comment:
None of this has anything to do with initialization and should not be included in the test.

Reply:
This test is to see whether the Sync can work well with the changes of sync(). Is it ok to only check whether that call of Sync() will get no error?

Reply:
Yeah, I think it's fine to only test what happens during initialization with that option set, doing this unstable test isn't worth it imo.